plugins/ingest/src/test/java/org/elasticsearch/ingest/PipelineTests.java

```diff
@@ -0,0 +1,52 @@
+package org.elasticsearch.ingest;
+
+import org.elasticsearch.common.xcontent.support.XContentMapValues;
+import org.elasticsearch.ingest.processor.Processor;
+import org.elasticsearch.ingest.processor.add.AddProcessor;
+import org.elasticsearch.ingest.processor.remove.RemoveProcessor;
+import org.elasticsearch.test.ESTestCase;
+import org.hamcrest.Matchers;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class PipelineTests extends ESTestCase {
+
+    public void testProcessorSettingsRemainUntouched() throws Exception {
+        Map<String, Object> subField = new HashMap<>();
+        subField.put("_subfield", "value");
+        Map<String, Object> fieldSettings = new HashMap<>();
+        fieldSettings.put("_field", subField);
+        Map<String, Object> addSettings = new HashMap<>();
+        addSettings.put("fields", fieldSettings);
+        Map<String, Object> removeSettings = new HashMap<>();
+        removeSettings.put("fields", Collections.singletonList("_field._subfield"));
+        Pipeline pipeline = createPipeline(processorConfig(AddProcessor.TYPE, addSettings), processorConfig(RemoveProcessor.TYPE, removeSettings));
+
+        IngestDocument ingestDocument = new IngestDocument("_index", "_type", "_id", new HashMap<>());
+        pipeline.execute(ingestDocument);
+
+        assertThat(XContentMapValues.extractValue("_field", ingestDocument.getSource()), Matchers.notNullValue());
```

Comment:
I was looking for a quick way to get and compare a value :) I'll change it to a more verbose casting to maps.